src/backend/utils/misc/guc.c

```diff
@@ -12216,7 +12216,36 @@ ProcessGUCArray(ArrayType *array, GucSource source)
 		 * GPSQL needs to dispatch the database/user config to segments.
 		 */
 		if (Gp_role == GP_ROLE_DISPATCH)
-			appendStringInfo(&MyProcPort->override_options, "-c %s=%s ", name, value);
+		{
```

Comment:
It's better to make a function to handle this common logic which process the special char.